### PR TITLE
DeviceInputSystem: Add check for matchMedia in WebDeviceInputSystem

### PR DESCRIPTION
**Pull request by PolygonalSun**
A user in the forum recently found an issue where the error `matchMedia is not defined` was occurring in an offscreen canvas.  Because the offscreen canvas was in a worker thread, it had no context of the `Window` interface and the `matchMedia` function.  This PR will add a check in the `WebDeviceInputSystem` to verify the existence of this function before attempting to execute it.

Forum Link: https://forum.babylonjs.com/t/offscreen-canvas-matchmedia/30682/10

### CODE REVIEW

File: packages/dev/core/src/DeviceInput/InputDevices/webDeviceInputSystem.ts
Comment (bghgary) [inline packages/dev/core/src/DeviceInput/InputDevices/webDeviceInputSystem.ts:234]: `globalThis` is not supported in IE. Either we wait until David's PR is merged or use something different.

Comment (PolygonalSun) [reply packages/dev/core/src/DeviceInput/InputDevices/webDeviceInputSystem.ts:234]: I just checked `typeof matchMedia === "function"` and it does work in a worker thread.  When I initially tried just using just `matchMedia`, an Uncaught ReferenceError was thrown, which is why I used `globalThis`.  I have no problem just changing this to use typeof since it works in the same way.

**Review by deltakosh — APPROVED** [review]

**Pull request merged**
